client/src/tests/relatedProductsTests.test.js

```diff
@@ -32,33 +44,111 @@ describe('RelatedProducts', () => {
+      let product = instance.product()
+      let style = instance.style()
+
+     return  Promise.all([
+        product,
+        style
+      ]).then(results => {
+        // console.log(results[1])
+          let resultStyleWithId = [];
+          results[0].forEach((product, pi) => {
+            results[1].forEach((style, si) => {
+
+              if (pi === si) {
+                resultStyleWithId.push(helper.addIdToStylesData(style, product.id))
+
+              }
+            })
+          })
+
+          let allPropsObj = helper.compileRelatedProductsDataToProps(results[0], resultStyleWithId)
+          let outfitPropsObj = helper.compileYourOutfitDataToProps(props.state.productInformation, resultStyleWithId[0])
+          instance.setState({
+            relatedProducts: results[0],
+            relatedProductsStyles: resultStyleWithId,
+            allPropsObj : allPropsObj,
+            outfitPropsObj: outfitPropsObj
+          })
+
+          // console.log(instance)
+      })
 
     })
     //unfinished new test for componentDidMount here - this one is for rendering
     test('on mount, component displays correct styles for current related product', () => {
+
+
       const component = renderer.create(<RelatedProducts state={props.state}/>)
       const instance = component.getInstance()
-      // console.log(instance)
-      // console.log(component.toJSON())
-      //for whatever reason you the instance log above does not show the state.
-      //however if you run this test - and console.log(this.state) in your component
-      //after the state has been set - it shows up.  This is super weird because in my tests
-      //im able to see the new state and also all the nodes it generates
 
-      //in my tests i have to mock out my ajax function in component did mount before i can see state update
-      //in yours - it populates with wrapoper but not renderer.create
+      let product = instance.product()
+      let style = instance.style()
+      return  Promise.all([
+        product,
+        style
+      ]).then(results => {
+        // console.log(results[1])
+          let resultStyleWithId = [];
+          results[0].forEach((product, pi) => {
+            results[1].forEach((style, si) => {
+
+              if (pi === si) {
+                resultStyleWithId.push(helper.addIdToStylesData(style, product.id))
+
+              }
+            })
+          })
+
+          let allPropsObj = helper.compileRelatedProductsDataToProps(results[0], resultStyleWithId)
+          let outfitPropsObj = helper.compileYourOutfitDataToProps(props.state.productInformation, resultStyleWithId[0])
+          instance.setState({
+            relatedProducts: results[0],
+            relatedProductsStyles: resultStyleWithId,
+            allPropsObj : allPropsObj,
+            outfitPropsObj: outfitPropsObj
+          })
+
+          console.log(component.toJSON().children[0].children[1].children[0].children[0].children)
```

Comment:
Super excited to dive into adding some more tests here. Crazy how complicated it was the get this set up, but I guess could be expected given the number of API calls my component is making

Reply:
glad someone appreciated it 🔥